stellargraph/core/graph.py

```diff
@@ -18,13 +18,65 @@
 a machine-learning ready graph used by models.
 
 """
-__all__ = ["StellarGraph", "StellarDiGraph", "GraphSchema"]
+__all__ = ["StellarGraph", "StellarDiGraph", "GraphSchema", "NeighbourWithWeight"]
 
 from typing import Iterable, Any, Mapping, List, Optional, Set
+from collections import namedtuple
+import pandas as pd
+import numpy as np
+import scipy.sparse as sps
 
 from .. import globalvar
-from .schema import GraphSchema
+from .schema import GraphSchema, EdgeType
 from .experimental import experimental
+from .element_data import NodeData, EdgeData, ExternalIdIndex
+from .utils import is_real_iterable
+from .validation import comma_sep
+
+
+NeighbourWithWeight = namedtuple("NeighbourWithWeight", ["node", "weight"])
+
+_WeightConfig = namedtuple("_WeightConfig", ["default", "label"])
+
+
+def _convert_single(type_name, data, *, shared_cols, weight_config, parent_name):
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError(
+            f"{parent_name}[{type_name!r}]: expected pandas DataFrame, found {type(data)}"
+        )
+
+    if weight_config is not None:
+        if weight_config.label not in data.columns:
+            data = data.assign(**{globalvar.WEIGHT: weight_config.default})
+        else:
+            data = data.rename(columns={weight_config.label: globalvar.WEIGHT})
+
+    shared = data[shared_cols]
```

Comment:
Ah, hm. It does need to be shared. I think the concept of `shared_cols` is now outdated, because the underlying `element_data` types use fixed names from `globalvar`, so really this all needs to do renaming similar to weights.